### PR TITLE
switch to solcjs, fix formatting unit test, update chalk dependency

### PR DESCRIPTION
**Pull request by linusnorton**
Unfortunately I also had to update chalk to get it to build correctly.

### CODE REVIEW

File: lib/abiParser.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import debug from "./debug";
 import { EvmType, VoidType, parseEvmType } from "./typeParser";
-import { yellow } from "chalk";
+import chalk from "chalk";
 import { MalformedAbiError } from "./errors";
 
+const { yellow } = chalk;
```
Comment (krzkaczor) [inline lib/abiParser.ts]: please revert. Why do you prefer object destruction instead doing this on one go while importing?

File: lib/cli.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env node
 import { readFileSync, writeFileSync } from "fs";
-import { blue, red, green } from "chalk";
+import chalk from "chalk";
```
Comment (krzkaczor) [inline lib/cli.ts]: same

File: lib/cli.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ import { parseArgs } from "./parseArgs";
 import { copyRuntime } from "./copyRuntime";
 import { extractAbi } from "./abiParser";
 
+const { blue, red, green } = chalk;
```
Comment (krzkaczor) [inline lib/cli.ts]: same

File: package.json [hunk 4]
```diff
@@ -31,12 +31,14 @@
     "web3-typescript-typings": "^0.7.2"
   },
   "dependencies": {
-    "chalk": "^2.1.0",
+    "bignumber.js": "^5.0.0",
```
Comment (krzkaczor) [inline package.json]: bignumber should be peer dependency

File: test/integration/ContractWithOverloads.spec.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { expect } from "chai";
 import { deployContract } from "./utils/web3Contracts";
 
 import { web3 } from "./web3";
-import { ContractWithOverloads } from "./abis/ContractWithOverloads";
+import { __ContractWithOverloads_sol_ContractWithOverloads as ContractWithOverloads } from "./abis/__ContractWithOverloads_sol_ContractWithOverloads";
```
Comment (krzkaczor) [inline test/integration/ContractWithOverloads.spec.ts]: isn't there a way to make solcjs generate more readable file names?

Comment (krzkaczor) [reply test/integration/ContractWithOverloads.spec.ts]: It pollutes a lot of code otherwise

File: lib/abiParser.ts (hunk 1 above)
Comment (linusnorton) [reply lib/abiParser.ts]: Unfortunately that format is no longer supported, see https://github.com/chalk/chalk/issues/215#issuecomment-338889272

File: test/integration/ContractWithOverloads.spec.ts (hunk 5 above)
Comment (linusnorton) [reply test/integration/ContractWithOverloads.spec.ts]: I didn't find one. I `cd` into that directory to stop it generating even more noise. I will dig a bit more and get back to you.

File: lib/abiParser.ts (hunk 1 above)
Comment (krzkaczor) [reply lib/abiParser.ts]: Uhh, that's horrifying but I guess we need to roll with it 👍

Comment (linusnorton) [reply lib/abiParser.ts]: Yeah, sorry about that. The maintainer of chalk seems quite adamant that it is the best way to do it :/

File: package.json (hunk 4 above)
Comment (linusnorton) [reply package.json]: I think it should be both as the project needs it and downstream projects need. Is that how peerDependencies work? You've got web3 as both a dependency and a peerDependency so I guess so.

**Comment by linusnorton** [comment]
@krzkaczor I think this is as ready as it's ever going to be. I didn't find a way to tidy up the contract filenames (apart from manually renaming them).

**Comment by krzkaczor** [comment]
@linusnorton thanks! Sorry, i was busy. I will make a review of this and the other branch before monday!

**Pull request merged**